src/datagrid/DataGrid.tsx

```diff
@@ -0,0 +1,400 @@
+    pagination?: boolean;
+    columns: DataGridColumn[];
+    data: DataGridRow[];
+    footer?: DataGridFooter;
+    onRowSelect?: Function;
+    onSelectAll?: Function;
+};
+
+/**
+ * type for DataGridColumn :
+ * @param {content} column data
+ * @param {sort} does colum support sorting
+ * @param {filter} does colum support filtering
+ * @param {className} CSS class name
+ * @param {columns} column details
+ * @param {style} CSS style
+ * @param {onFilter} Function for custom filtering
+ */
+type DataGridColumn = {
+    content: any;
+    sort?: boolean;
+    filter?: boolean;
+    className?: string;
+    style?: any;
+    onFilter?: Function;
+};
+
+/**
+ * type for DataGridRow :
+ * @param {content} row data
+ * @param {rowID} unique ID to identify row
+ * @param {isSelected} set to true if row is selected
+ * @param {className} CSS class name
+ * @param {style} CSS style
+ */
+type DataGridRow = {
+    content: DataGridCell[];
+    className?: string;
+    style?: any;
+    rowID?: string;
+    isSelected?: boolean;
+};
+
+/**
+ * type for DataGridCell :
+ * @param {content} cell data
+ * @param {className} CSS class name
+ * @param {style} CSS style
+ */
+type DataGridCell = {
+    content: any;
+    className?: string;
+    style?: any;
+};
+
+/**
+ * type for DataGridFooter :
+ * @param {content} Footer data
+ * @param {className} CSS class name
+ * @param {style} CSS style
+ */
+type DataGridFooter = {
+    content: any;
+    className?: string;
+    style?: any;
+};
+
+/**
+ * Enum for GridSelectionType :
+ * @param {MULTI} for enabling multi row select
+ * @param {SINGLE} for enabling single row select
+ */
+export enum GridSelectionType {
+    MULTI = "multi",
+    SINGLE = "single",
+}
+
+/**
+ * State for DataGrid :
+ * @param {selectAll} set to true if all rows got selected else false
+ * @param {allColumns} column data
+ * @param {allRows} row data
+ */
+type DataGridState = {
+    selectAll: boolean;
+    allColumns: DataGridColumn[];
+    allRows: DataGridRow[];
+};
+
+/**
+ * DataGrid Componnet :
+ * Displays data in grid format
+ */
+export class DataGrid extends React.PureComponent<DataGridProps, DataGridState> {
+    // Default props for datagrid
+    static defaultProps = {
+        pagination: false,
+    };
+
+    // Initial state of datagrid
+    state: DataGridState = {
+        selectAll: false,
+        allColumns: this.props.columns,
+        allRows: this.props.data,
+    };
+
+    /* ##########  DataGrid public methods start  ############ */
+    // Function to return all selected rows
+    getSelectedRows(): DataGridRow[] {
+        const {allRows} = this.state;
+        let selectedRows = new Array();
+        if (this.state.selectAll) selectedRows = allRows;
+        else {
+            allRows.forEach(row => {
+                if (row["isSelected"]) selectedRows.push(row);
+            });
+        }
+        return selectedRows;
+    }
+
+    /* ##########  DataGrid public methods end  ############ */
```

Comment:
same. exceeded comment

Reply:
probably is to have just a convention
For example
- class Name
- statis fields
- state
- local fields
- constructor
- public methods
- private methods

Reply:
done